md5/src/lib.rs

```diff
@@ -124,6 +129,38 @@ impl fmt::Debug for Md5Core {
     }
 }
 
+impl SerializableState for Md5Core {
+    type SerializedStateSize = U24;
+
+    fn serialize(&self) -> SerializedState<Self> {
+        let mut serialized_state = GenericArray::<u8, U16>::default();
+
+        for (i, bytes) in serialized_state.as_mut_slice().chunks_mut(4).enumerate() {
+            bytes.copy_from_slice(self.state[i].to_be_bytes().as_slice());
+        }
```

Comment:
In Rust it's usually preferred to `zip` iterators in such cases instead of using `enumerate`.
```rust
let mut state = GenericArray::<u8, U16>::default();
for (val, chunk) in self.state.iter().zip(state.chunks_exact_mut(4)) {
    chunk.copy_from_slice(&val.to_le_bytes());
}
// You also can transform arrays into GenericArray and back using `From`/`Into`
state.concat(self.block_len.to_le_bytes().into());
```

Also little-endian would be more efficient on most currently used CPUs, so I think we should prefer it. (BE is used traditionally by cryptographers, by it begins to change slowly).